### PR TITLE
content of specific file in multifile gist

### PR DESCRIPTION
**Pull request by grizmin**
content of specific file in multifile gist

### CODE REVIEW

**Comment by jdowner** [comment]
That for adding this feature. Before I can accept the PR there are a few changes I'd like to ask you to make:

1. squash the commits
2. all the tests should pass
3. the new functionality needs to support decryption
4. rather than iterate over all the files, just do a lookup in the dict
5. add an example to the 'help'

**Comment by grizmin** [comment]
Done.

File: bin/gist [hunk 1]
```diff
@@ -350,6 +357,11 @@ def main(argv=sys.argv[1:], config=None):
         logger.debug(u'action: - {}'.format(gist_id))
 
         content = gapi.content(gist_id)
+        if args['<filename>'] in content.keys():
+            gist_file = args['<filename>']
+        else:
+            gist_file = False
```
Comment (jdowner) [inline bin/gist]: I would prefer
```
gist_file = content.get('<filename>')
```
which will be 'None' if no filename has been provided.

File: bin/gist [hunk 2]
```diff
@@ -358,12 +370,19 @@ def main(argv=sys.argv[1:], config=None):
             logger.debug(u'action: - {}'.format(homedir))
 
             gpg = gnupg.GPG(gnupghome=homedir, use_agent=True)
+            if gist_file:
```
Comment (jdowner) [inline bin/gist]:
```
if gist_file is not None:
```

File: bin/gist [hunk 3]
```diff
@@ -358,12 +370,19 @@ def main(argv=sys.argv[1:], config=None):
...
+                print(gpg.decrypt(content[gist_file]).data.decode('utf-8'))
+                return
 
             for name, lines in content.items():
                 lines = gpg.decrypt(lines).data.decode('utf-8')
                 print(u'{} (decrypted):\n{}\n'.format(name, lines))
 
         else:
+            if gist_file:
```
Comment (jdowner) [inline bin/gist]:
```
if gist_file is not None:
```

Comment (jdowner) [inline bin/gist]: Also fix the typo in the commit message.

File: bin/gist [hunk 4]
```diff
@@ -95,6 +95,13 @@ Commands:
         followed by a colon, and then the content of that file is written to
         the terminal.
 
+        If filename is given, only the content of the specified filename
```
Comment (jdowner) [inline bin/gist]: "If a filename...."

**Comment by grizmin** [comment]
Thank you for the time spent. It should be more pythonish already :)

**Comment by jdowner** [comment]
Thanks Konstantin!

**Pull request merged**